Automattic/wp-calypso
# WIP: Generic email components - `EmailAccountsCard` and `EmailAccountItem`

Also introduces the concept of 'preparing' accounts in global state.

#### Changes proposed in this Pull Request

* A `prepareAccounts` data processor that uses provider specific logic to create items attached to mailboxes just as soon as the data is received from APIs. This will reduce the need to use logici specific to provider in each component that requests this data.
* A new `EmailAccountsCard` component to list accounts and mailboxes irrespective of provider
* A new `EmailAccountItem` to list mailboxes associated with accounts

#### Testing instructions

- Startup your dev server
- Navigate to [/email](http://calypso.localhost:3000/email/)
- Confirm that you see items comparable to the accounts in the `emailAccounts` key in global state.
